src/game/etj_timerun_v2.cpp

```diff
@@ -252,6 +252,10 @@ void ETJump::TimerunV2::clientConnect(int clientNum, int userId) {
         auto clientConnectResult =
             dynamic_cast<ClientConnectResult *>(result.get());
 
+        if (!clientConnectResult) {
+          throw std::runtime_error("clientConnectResult is NULL");
```

Comment:
Think we could technically just assert this and crash the application as this should never happen but should be fine this way too